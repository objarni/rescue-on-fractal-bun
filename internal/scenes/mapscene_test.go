package scenes

import (
	"fmt"
	"github.com/faiface/pixel"
	"objarni/rescue-on-fractal-bun/internal"
)

func LocAt(x float64, y float64) internal.MapSign {
	return internal.MapSign{
		MapPos: pixel.Vec{X: x, Y: y},
	}
}

func ExampleFindNearLocation_tooFarAwayMeansNotClose() {
	ix := FindNearMapSign(
		pixel.Vec{X: 99, Y: 99},
		[]internal.MapSign{
			LocAt(1000, 1000),
		},
		30,
	)
	fmt.Println("Index of closest location:", ix)
	// Output:
	// Index of closest location: -1
}
