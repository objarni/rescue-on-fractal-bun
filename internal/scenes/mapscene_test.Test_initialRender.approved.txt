WinOp Sequence:
  Moved 400 pixels right 300 pixels up:
    Image "IMap"
  WinOp from ImdOp:
    ImdOp Sequence:
      ImdOp Sequence:
        Color 72, 61, 139:
          Circle radius 5 center <246, 109> thickness 3
        Color 72, 61, 139:
          Circle radius 5 center <355, 235> thickness 3
        Color 72, 61, 139:
          Circle radius 5 center <299, 375> thickness 3
        Color 0, 128, 0:
          Circle radius 15 center <299, 375> thickness 3
        Color 255, 0, 0:
          Circle radius 11 center <299, 375> thickness 3
      Color 255, 0, 0:
        ImdOp Sequence:
          Line from <299, 0> to <299, 600> thickness 2
          Line from <0, 375> to <800, 375> thickness 2